R/methods-epi_df.R

```diff
@@ -153,7 +164,30 @@ dplyr_reconstruct.epi_df <- function(data, template) {
   # keep any grouping that has been applied:
   res <- NextMethod()
 
-  col_names <- names(res)
+  reconstruct_light_edf(res, template)
+}
+
+#' Like `dplyr_reconstruct.epi_df` but not recomputing any grouping
+#'
+#' In the move to our current not-quite-proper/effective "implementation" of
+#' [`dplyr::dplyr_extending`] for `epi_df`s, we moved a lot of checks in
+#' `dplyr_reconstruct` and used it instead of [`reclass()`] in various
+#' operations to prevent operations from outputting invalid metadata/classes,
+#' instead of more careful tailored and relevant checks. However, this actually
+#' introduced extra overhead due to `dplyr_reconstruct.epi_df()` passing off to
+#' `dplyr_reconstruct.grouped_df()` when grouped, which assumes that it will
+#' need to / should for safety recompute the groups, even when it'd be safe for
+#' it not to do so. In many operations, we're using `NextMethod()` to dispatch
+#' to `grouped_df` behavior if needed, and it should output something with valid
+#' groupings.
```

Comment:
Thanks for this description, this is v handy. Also good catch on finding this intervention!